core/embed/sec/secret/stm32f4/secret.c

```diff
@@ -84,18 +84,24 @@ void secret_unlock_bootloader(void) { secret_erase(); }
 void secret_write_header(void) {
   uint8_t header[SECRET_HEADER_LEN] = {0};
   memcpy(header, SECRET_HEADER_MAGIC, SECRET_HEADER_MAGIC_LEN);
-  secret_write(header, SECRET_HEADER_OFFSET, SECRET_HEADER_LEN);
+  ensure(secret_write(header, SECRET_HEADER_OFFSET, SECRET_HEADER_LEN),
+         "secret write header failed");
 }
 
-void secret_write(const uint8_t* data, uint32_t offset, uint32_t len) {
+secbool secret_write(const uint8_t* data, uint32_t offset, uint32_t len) {
```

Comment:
I just want to have it confirmed that I understand how to write the certificate. I assume that:
  * I should use `secret_write()`.
  * It's my responsibility to check that `len <= SECRET_MCU_DEVICE_CERT_SIZE`.

Is that correct? Can `len` be less than `SECRET_MCU_DEVICE_CERT_SIZE`?

Reply:
- yes use secret_write
- yes currently the API is quite dumb, we might want to improve it in future
- yes it can be less, though there is a limitation that you have to use data with size being multiple of 16 bytes. I can also remove this limitation if it causes issues.